tests/python/contrib/test_ethosu/test_codegen.py

```diff
@@ -349,6 +349,7 @@ def representative_dataset():
         ([1, 2, 3, 4], [1, 2, 3, 4]),
         ([1, 2, 3, 4], [1, 1, 1, 1]),
         ([1, 1, 1, 1], [1, 2, 3, 4]),
+        ([1, 4, 4], [4, 1]),
```

Comment:
Other cases worth testing are input shape that is less than 4D and the first dimension is not 1 and 2D and 1D IFMs

Reply:
I was being a bit wary here about adding too many tests as adding all 3 cases would lead to 120 additional tests rather than just 40 :) I wonder if its worth testing this separately for just `ADD` for example as the underlying operation doesn't matter for this change?

Reply:
Yes, good point, 120 tests is a bit too much :D Yes, I think it is a good idea to test a few IFMs with all the "flavours" of the binary elementwise operators and then have a separate test just for one of them to test all the corner cases